src/compiler/diagnosticMessages.json

```diff
@@ -127,11 +127,27 @@
         "category": "Error",
         "code": 1039
     },
+    "'{0}' modifier cannot be used in an ambient context.": {
+        "category": "Error",
+        "code": 1040
+    },
+    "'{0}' modifier cannot be used with a class declaration.": {
+        "category": "Error",
+        "code": 1041
+    },
+    "'{0}' modifier cannot be used here.": {

```

Comment:
I’m using this following a suggestion from Cyrus, and also following the behavior of the C# compiler here.

From: Jason Freeman [mailto:notifications@github.com]
Sent: Thursday, February 12, 2015 2:25 PM
To: Microsoft/TypeScript
Cc: Ron Buckton
Subject: Re: [TypeScript] Async Functions (#1781)


In src/compiler/diagnosticMessages.json<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24626698>:

> @@ -127,11 +127,27 @@

>          "category": "Error",

>          "code": 1039

>      },

> +    "'{0}' modifier cannot be used in an ambient context.": {

> +        "category": "Error",

> +        "code": 1040

> +    },

> +    "'{0}' modifier cannot be used with a class declaration.": {

> +        "category": "Error",

> +        "code": 1041

> +    },

> +    "'{0}' modifier cannot be used here.": {

When would you use this one? Seems like you could use it anywhere. Also, will the user understand why they can't use something 'here'?

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24626698>.